core/blockchain.go

```diff
@@ -989,9 +913,13 @@ func (bc *BlockChain) stopWithoutSaving() {
 	if !bc.stopping.CompareAndSwap(false, true) {
 		return
 	}
+	// Signal shutdown tx indexer.
+	if bc.txIndexer != nil {
+		bc.txIndexer.close()
+	}
 
 	log.Info("Closing quit channel")
-	close(bc.quit)
+	close(bc.quit) // XXX: Why is this closed before [scope] but it's opposite upstream?
```

Comment:
Why is `bc.quit` closed before scope but it's opposite upstream?

Reply:
I'm leaving it for now as is and creating this issue to look into it later https://github.com/ava-labs/subnet-evm/issues/1283